packages/zoe/src/zoeService/escrowStorage.js

```diff
@@ -75,28 +81,35 @@ export const makePurseStorage = () => {
     const wantKeywords = Object.keys(want);
     const proposalKeywords = harden([...giveKeywords, ...wantKeywords]);
 
+    // If any of these deposits hang or fail, then depositPayments
+    // hangs or fails, the offer does not succeed, and any funds that
+    // were deposited into the pool purses are lost. We have a ticket
+    // for giving the user a refund of what was already deposited, and
+    // offer safety and payout liveness are still meaningful as long
+    // as issuers are well-behaved. For more, see
+    // https://github.com/Agoric/agoric-sdk/issues/1271
     const amountsDeposited = await Promise.all(
       giveKeywords.map(keyword =>
         doDepositPayments(payments[keyword], give[keyword]),
       ),
     );
 
-    const amountsWanted = wantKeywords.map(keyword =>
+    const emptyAmountsForWantKeywords = wantKeywords.map(keyword =>
       amountMath.makeEmptyFromAmount(want[keyword]),
     );
 
     // Note: payments without a matching `give` keyword are ignored.
     const initialAllocation = arrayToObj(
-      [...amountsDeposited, ...amountsWanted],
+      [...amountsDeposited, ...emptyAmountsForWantKeywords],
       proposalKeywords,
     );
 
     return initialAllocation;
   };
 
   return {
-    createPurse,
-    createLocalPurse,
+    createPurse, // create does not return
```

Comment:
```js
// create does not return anything
```
The current phrase implies that, like `assert.fail`, control flow does not proceed normally after this point.

Reply:
oo good catch

Reply:
fixed